awswrangler/s3/_read_parquet.py

```diff
@@ -331,6 +331,86 @@ def _read_parquet(  # pylint: disable=W0613
     return _utils.table_refs_to_df(tables, kwargs=arrow_kwargs)
 
 
+def read_parquet_chunked(
+    path: Union[str, List[str]],
+    path_root: Optional[str] = None,
+    dataset: bool = False,
+    path_suffix: Union[str, List[str], None] = None,
+    path_ignore_suffix: Union[str, List[str], None] = None,
+    ignore_empty: bool = True,
+    partition_filter: Optional[Callable[[Dict[str, str]], bool]] = None,
+    columns: Optional[List[str]] = None,
+    validate_schema: bool = False,
+    coerce_int96_timestamp_unit: Optional[str] = None,
+    last_modified_begin: Optional[datetime.datetime] = None,
+    last_modified_end: Optional[datetime.datetime] = None,
+    version_id: Optional[Union[str, Dict[str, str]]] = None,
+    chunk_size: Optional[int] = None,
+    use_threads: Union[bool, int] = True,
+    boto3_session: Optional[boto3.Session] = None,
+    s3_additional_kwargs: Optional[Dict[str, Any]] = None,
+    pyarrow_additional_kwargs: Optional[Dict[str, Any]] = None,
+) -> Iterator[pd.DataFrame]:
+    paths: List[str] = _path2list(
```

Comment:
Ideally, we would want to refactor this in a way where we don't need to copy paste these 40-50 lines.